homeassistant/components/spider/sensor.py

```diff
@@ -0,0 +1,140 @@
+"""Support for Spider Powerplugs (energy & power)."""
+from datetime import date, datetime
+
+from homeassistant.components.sensor import STATE_CLASS_MEASUREMENT, SensorEntity
+from homeassistant.const import (
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_POWER,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+)
+from homeassistant.helpers.entity import DeviceInfo
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config, async_add_entities):
+    """Initialize a Spider Power Plug."""
+    api = hass.data[DOMAIN][config.entry_id]
+    async_add_entities(
+        [
+            SpiderPowerPlugEnergy(api, entity)
+            for entity in await hass.async_add_executor_job(api.get_power_plugs)
+        ]
+    )
+    async_add_entities(
+        [
+            SpiderPowerPlugPower(api, entity)
+            for entity in await hass.async_add_executor_job(api.get_power_plugs)
+        ]
+    )
+
+
+class SpiderPowerPlugEnergy(SensorEntity):
+    """Representation of a Spider Power Plug (energy)."""
+
+    def __init__(self, api, power_plug) -> None:
+        """Initialize the Spider Power Plug."""
+        self.api = api
+        self.attr_name = f"{power_plug.name} Total Energy"
+        self.id = f"{power_plug.id}_total_energy"
+        self.power_plug = power_plug
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device_info of the device."""
+        return {
+            "identifiers": {(DOMAIN, self.id)},
```

Comment:
Same comment as above, this is the entity unique ID used as device ID, should a device be more global, with multiple entities?

I expected `power_plug.id` to be used here